org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/xpect/ConstructorDelegation.wlk.xt

```diff
@@ -50,7 +50,7 @@ class SuperClass {
 }
 class SubClass inherits SuperClass { 
 	var anotherVariable
-	// XPECT warnings --> "Redundant super constructor call. Default constructor is not overriden." at "super()"
+	// XPECT warnings --> "Invalid super constructor call. Superclass has no explicit constructors." at "super()"
```

Comment:
Why do we need a specific message, I think that the standard one should be good enough: 
"Invalid super constructor call. It does not exist in superclass. Valid constructors are: super()"

Also, is there a test for A inherits B inherits C where superclass B does not define constructors but granpa C does?

Reply:
I don't remember why I changed Redundant to Invalid, but the error is different.

If you do

```javascript
class SuperClass {
	var superX
	
	method getSuperX() { return superX }
	method setSuperX(value) { superX = value }
}
class SubClass inherits SuperClass { 
	var anotherVariable
	constructor(n) = super() {
		anotherVariable = n
	}
	method getAnotherVariable() = anotherVariable
}
```

The warning would show: "Valid constructors are: super()". "Ok", should think the programmer, "I'm calling right to super constructor". But what you have to do is delete the delegating super constructor call and use your plain definition:

```javascript
class SuperClass {
	var superX
	
	method getSuperX() { return superX }
	method setSuperX(value) { superX = value }
}
class SubClass inherits SuperClass { 
	var anotherVariable
	constructor(n) {
		anotherVariable = n
	}
	method getAnotherVariable() = anotherVariable
}
```

So the problem is not with an "invalid" constructor, but with an innecessary step.

Reply:
And regarding the second question: 
"is there a test for A inherits B inherits C where superclass B does not define constructors but granpa C does?"

Yes, 
ConstructorTest.xtend
See multipleInheritanceForConstructors4 test.